### ISSUE TITLE
panel shows unavailable need to reload the integration 

### ISSUE THREAD

**Issue by This-is-me-yes** [hawky358/hass_ids_hyyp#11]
ive found a few occasions that the panel shows unavailable need to reload the integration - Functions work but the live info shows unavailable:

<img width="1410" alt="Screenshot 2023-06-18 at 12 10 36" src="https://github.com/hawky358/hass_ids_hyyp/assets/136916115/242cf372-516b-43d2-9488-dd5c24ab14c0">

**Comment by hawky358** [comment]
- Can you please check the system logs to see if there are any errors?
- Does it become unavailable "permanently" or does it fix itself after ~30-60 seconds? This may indicate a timeout or IDS server delay of some sort
- Did you reboot HASS or is there some internet issue related to the timings?


I'm busy testing a build which can dump debug logs which will help trace issues. Will publish as a test branch in the next few days.

**Comment by This-is-me-yes** [comment]
I reload the HYYP integration and all the "info" then goes back to normal from "unavailable". No online issues .. enabled the debug and will let you know when I have info ... while in the "unavailable" state,  the panel arms and functions but the notifications don't arrive so the keypad state does not change ... my IDS app does get the notifications and functions properly 

It stays unavailable and the only fix is to reload the HYYP integration.... 


<img width="1209" alt="Screenshot 2023-06-18 at 16 40 23" src="https://github.com/hawky358/hass_ids_hyyp/assets/136916115/539e2b56-a9c1-44fc-8f79-4beca9cb8bd4">

**Comment by This-is-me-yes** [comment]
![Uploading Screenshot 2023-06-18 at 16.53.07.png…]()

**Comment by This-is-me-yes** [comment]
<img width="1440" alt="Screenshot 2023-06-18 at 16 53 07" src="https://github.com/hawky358/hass_ids_hyyp/assets/136916115/ba7bd8ad-a3cb-43c4-b840-b579ae49c86b">

**Comment by This-is-me-yes** [comment]
just updated the to latest release with debug will let you know what it finds

**Comment by This-is-me-yes** [comment]
<img width="961" alt="Screenshot 2023-06-18 at 17 03 07" src="https://github.com/hawky358/hass_ids_hyyp/assets/136916115/09e909b5-9ca4-4b83-933c-166a48c3b3ec">

**Comment by hawky358** [comment]
1. Please enable debug logging on IDS integration. 
2. Let it run for a a bit. When you click disable it will download a log file which you can send to me
3. At the Settings > Logs. Please clear the logs then check if there are any errors showing up. You don't have to open "Full logs"

The  1st screenshot shows something around the bypass sensors
The 2nd screenshot only shows feedback from IDS. It could be that the pin isn't entered or the system is already disarmed, or there was some sort of timeout.

Try 1,2 and 3. I'll check if I can see something in the logs.

**Comment by hawky358** [comment]
> 3. At the Settings > Logs. Please clear the logs then check if there are any errors showing up. 

You'll have to restart Home assistant after clearing so everything can start up again.

**Comment by This-is-me-yes** [comment]
> 

Yes its sending the disarm command as the status did not update so my HA and Alarm are out of sync.. all clear and reset logs and get back to u

**Comment by This-is-me-yes** [comment]
<img width="1440" alt="Screenshot 2023-06-18 at 17 28 16" src="https://github.com/hawky358/hass_ids_hyyp/assets/136916115/07451dfa-676e-4ea7-b184-4bb0e5c82e2a">

**Comment by This-is-me-yes** [comment]
Logger: custom_components.ids_hyyp.coordinator
Source: helpers/update_coordinator.py:229 
Integration: IDS Hyyp (hawkMod) ([documentation](https://github.com/hawky358/hass_ids_hyyp)) 
First occurred: 18:00:42 (1 occurrences) 
Last logged: 18:00:42

Error fetching ids_hyyp data: Invalid response from API: Error getting sync info from api: [{'code': 'unknown', 'message': 'An unexpected API error occurred', 'fields': None}]

**Comment by hawky358** [comment]
I need to figure out why your zones don't have the triggered key. It should be assigned to every zone. I need the debug log from the integration to try and figure something out. Please check below and also the questions:

1. Are you using a the IDS hyyp or the ADT Smarthome system?
2. Have you logged in as such on the integration? (za.co.adt.securehome.android  OR  com.hyyp247.home)
3. Please send me the log file by following the following steps: (Make sure you're running 1.3.2-b1)

- Open the integration settings (Gear button)
![image](https://github.com/hawky358/hass_ids_hyyp/assets/16265473/621f54e1-afc0-4ad4-98c9-d6d846f1f73c)

- Click Enable debug logging
![image](https://github.com/hawky358/hass_ids_hyyp/assets/16265473/29311b51-06f2-420b-a0b5-919552b6d9e8)

- When you go back it should look like this: 
![image](https://github.com/hawky358/hass_ids_hyyp/assets/16265473/ea23881a-555d-4f40-956d-5162d89c4bc6)

- Reboot Home assistant and let it run for around 5 minutes or so

- Go back to the settings and click disable debug logging. It will then allow you to download a .log file which I need you to send to me
![image](https://github.com/hawky358/hass_ids_hyyp/assets/16265473/a2cc3d9d-77fb-4963-8e31-5224b871cc6c)

**Comment by This-is-me-yes** [comment]
2023-06-18 17:00:38.347 DEBUG (MainThread) [custom_components.ids_hyyp.coordinator] INFOS:
2023-06-18 17:00:38.347 DEBUG (MainThread) [custom_components.ids_hyyp.coordinator] -----
2023-06-18 17:00:38.347 DEBUG (MainThread) [custom_components.ids_hyyp.coordinator] {240651: {'id': 240651, 'name': 'Home ', 'color': '#AAAFFF', 'partitionIds': [1887035], 'triggerIds': [36315], 'partitionIdsString': '1887035', 'triggerIdsString': '36315', 'hasPin': True, 'isMaster': True, 'isOnline': True, 'imei': '0188999370356', 'addressLine1': None, 'addressLine2': None, 'city': None, 'province': '', 'countryCode': 'ZA', 'areaCode': '0000', 'timeZoneOffset': 2, 'subscription': None, 'folderId': 5160, 'siteImageUrl': 'https://ids-test.trintel.co.za/Inhep-Impl-1.0-SNAPSHOT/../images/Site_240651_IMG_2675.PNG', 'dststartDayOfWeek': 1, 'dststopDayOfWeek': 1, 'dstenabled': False, 'dstoffset': 1, 'dststartMonth': 1, 'dststartWeek': 1, 'dststopMonth': 1, 'dststopWeek': 1, 'lastNoticeTime': '2023-06-18 16:29:43.674000', 'lastNoticeName': 'Stay arm', 'triggers': {7916: {'id': 7916, 'name': 'driveway gate', 'number': 49, 'type': 'pulse', 'onoff': False, 'color': None, 'hasPin': True, 'userId': 145}, 29637: {'id': 29637, 'name': 'Garage ', 'number': 42, 'type': 'pulse', 'onoff': False, 'color': None, 'hasPin': False, 'userId': 145}, 29639: {'id': 29639, 'name': 'Gate open', 'number': 1, 'type': 'switch', 'onoff': False, 'color': None, 'hasPin': False, 'userId': 145}, 32567: {'id': 32567, 'name': 'Gate open', 'number': 1, 'type': 'switch', 'onoff': False, 'color': None, 'hasPin': True, 'userId': 145}, 33835: {'id': 33835, 'name': 'Garage', 'number': 47, 'type': 'pulse', 'onoff': True, 'color': None, 'hasPin': True, 'userId': 145}, 36315: {'id': 36315, 'name': 'Gatemotor ', 'number': 5, 'type': 'pulse', 'onoff': False, 'color': None, 'hasPin': True, 'userId': 145}, 4716: {'id': 4716, 'name': 'Led', 'number': 5, 'type': 'pulse', 'onoff': None, 'color': None, 'hasPin': False, 'userId': 145}, 14124: {'id': 14124, 'name': 'GATE ', 'number': 52, 'type': 'pulse', 'onoff': False, 'color': None, 'hasPin': False, 'userId': 145}}, 'partitions': {1887035: {'id': 1887035, 'name': 'Partition 1', 'stayProfileIds': [252727], 'zoneIds': [4585759, 4585761, 4585763, 4585765, 4590729, 4590731, 4590733, 4590735], 'zoneIdsString': '4585759, 4585761, 4585763, 4585765, 4590729, 4590731, 4590733, 4590735', 'stayProfileIdsString': '252727', 'number': 1, 'hasPin': False, 'alarm': False, 'zones': {4585759: {'id': 4585759, 'name': 'ZONE1', 'number': 1, 'bypassed': False}, 4590729: {'id': 4590729, 'name': 'Zone 2', 'number': 2, 'bypassed': False}, 4585761: {'id': 4585761, 'name': 'ZONE3', 'number': 3, 'bypassed': False}, 4585763: {'id': 4585763, 'name': 'ZONE4', 'number': 4, 'bypassed': False}, 4585765: {'id': 4585765, 'name': 'ZONE5', 'number': 5, 'bypassed': False}, 4590731: {'id': 4590731, 'name': 'Zone 6', 'number': 6, 'bypassed': False}, 4590733: {'id': 4590733, 'name': 'Zone 7', 'number': 7, 'bypassed': False}, 4590735: {'id': 4590735, 'name': 'Zone 8', 'number': 8, 'bypassed': False}}, 'stayProfiles': {252727: {'id': 252727, 'name': 'Stay 1', 'zoneIds': [4590735, 4590733, 4590731], 'zoneIdsString': '4590735, 4590733, 4590731', 'number': 1}}, 'armed': True, 'stayArmed': True, 'stayArmedProfileName': 'Stay 1'}}}}
2023-06-18 17:00:38.348 DEBUG (MainThread) [custom_components.ids_hyyp.coordinator] -----
2023-06-18 17:00:38.348 DEBUG (MainThread) [custom_components.ids_hyyp.coordinator] </INFOS>
2023-06-18 17:00:38.348 DEBUG (MainThread) [custom_components.ids_hyyp.coordinator] Finished fetching ids_hyyp data in 2.459 seconds (success: True)

**Comment by This-is-me-yes** [comment]
Using ADT version ...za.co.adt.securehome.android

**Comment by This-is-me-yes** [comment]
za.co.adt.securehome.android using adt ....

2023-06-18 17:00:38.347 DEBUG (MainThread) [custom_components.ids_hyyp.coordinator] INFOS:
2023-06-18 17:00:38.347 DEBUG (MainThread) [custom_components.ids_hyyp.coordinator] -----
2023-06-18 17:00:38.347 DEBUG (MainThread) [custom_components.ids_hyyp.coordinator] {240651: {'id': 240651, 'name': 'Home ', 'color': '#AAAFFF', 'partitionIds': [1887035], 'triggerIds': [36315], 'partitionIdsString': '1887035', 'triggerIdsString': '36315', 'hasPin': True, 'isMaster': True, 'isOnline': True, 'imei': 'xxxxxxxxxxx', 'addressLine1': None, 'addressLine2': None, 'city': None, 'province': '', 'countryCode': 'ZA', 'areaCode': '0000', 'timeZoneOffset': 2, 'subscription': None, 'folderId': 5160, 'siteImageUrl': 'https://ids-test.trintel.co.za/Inhep-Impl-1.0-SNAPSHOT/../images/Site_240651_IMG_2675.PNG', 'dststartDayOfWeek': 1, 'dststopDayOfWeek': 1, 'dstenabled': False, 'dstoffset': 1, 'dststartMonth': 1, 'dststartWeek': 1, 'dststopMonth': 1, 'dststopWeek': 1, 'lastNoticeTime': '2023-06-18 16:29:43.674000', 'lastNoticeName': 'Stay arm', 'triggers': {7916: {'id': 7916, 'name': 'driveway gate', 'number': 49, 'type': 'pulse', 'onoff': False, 'color': None, 'hasPin': True, 'userId': 145}, 29637: {'id': 29637, 'name': 'Garage ', 'number': 42, 'type': 'pulse', 'onoff': False, 'color': None, 'hasPin': False, 'userId': 145}, 29639: {'id': 29639, 'name': 'Gate open', 'number': 1, 'type': 'switch', 'onoff': False, 'color': None, 'hasPin': False, 'userId': 145}, 32567: {'id': 32567, 'name': 'Gate open', 'number': 1, 'type': 'switch', 'onoff': False, 'color': None, 'hasPin': True, 'userId': 145}, 33835: {'id': 33835, 'name': 'Garage', 'number': 47, 'type': 'pulse', 'onoff': True, 'color': None, 'hasPin': True, 'userId': 145}, 36315: {'id': 36315, 'name': 'Gatemotor ', 'number': 5, 'type': 'pulse', 'onoff': False, 'color': None, 'hasPin': True, 'userId': 145}, 4716: {'id': 4716, 'name': 'Led', 'number': 5, 'type': 'pulse', 'onoff': None, 'color': None, 'hasPin': False, 'userId': 145}, 14124: {'id': 14124, 'name': 'GATE ', 'number': 52, 'type': 'pulse', 'onoff': False, 'color': None, 'hasPin': False, 'userId': 145}}, 'partitions': {1887035: {'id': 1887035, 'name': 'Partition 1', 'stayProfileIds': [252727], 'zoneIds': [4585759, 4585761, 4585763, 4585765, 4590729, 4590731, 4590733, 4590735], 'zoneIdsString': '4585759, 4585761, 4585763, 4585765, 4590729, 4590731, 4590733, 4590735', 'stayProfileIdsString': '252727', 'number': 1, 'hasPin': False, 'alarm': False, 'zones': {4585759: {'id': 4585759, 'name': 'ZONE1', 'number': 1, 'bypassed': False}, 4590729: {'id': 4590729, 'name': 'Zone 2', 'number': 2, 'bypassed': False}, 4585761: {'id': 4585761, 'name': 'ZONE3', 'number': 3, 'bypassed': False}, 4585763: {'id': 4585763, 'name': 'ZONE4', 'number': 4, 'bypassed': False}, 4585765: {'id': 4585765, 'name': 'ZONE5', 'number': 5, 'bypassed': False}, 4590731: {'id': 4590731, 'name': 'Zone 6', 'number': 6, 'bypassed': False}, 4590733: {'id': 4590733, 'name': 'Zone 7', 'number': 7, 'bypassed': False}, 4590735: {'id': 4590735, 'name': 'Zone 8', 'number': 8, 'bypassed': False}}, 'stayProfiles': {252727: {'id': 252727, 'name': 'Stay 1', 'zoneIds': [4590735, 4590733, 4590731], 'zoneIdsString': '4590735, 4590733, 4590731', 'number': 1}}, 'armed': True, 'stayArmed': True, 'stayArmedProfileName': 'Stay 1'}}}}
2023-06-18 17:00:38.348 DEBUG (MainThread) [custom_components.ids_hyyp.coordinator] -----
2023-06-18 17:00:38.348 DEBUG (MainThread) [custom_components.ids_hyyp.coordinator] </INFOS>
2023-06-18 17:00:38.348 DEBUG (MainThread) [custom_components.ids_hyyp.coordinator] Finished fetching ids_hyyp data in 2.459 seconds (success: True)
2023-06-18 17:00:38.387 ERROR (MainThread) [homeassistant.components.binary_sensor] Error adding entities for domain binary_sensor with platform ids_hyyp
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 471, in async_add_entities
await asyncio.gather(*tasks)
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 749, in _async_add_entity
await entity.add_to_platform_finish()
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 850, in add_to_platform_finish
self.async_write_ha_state()
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 590, in async_write_ha_state
self._async_write_ha_state()
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 654, in _async_write_ha_state
state = self._stringify_state(available)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 596, in _stringify_state
if (state := self.state) is None:
^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/components/binary_sensor/__init__.py", line 211, in state
if (is_on := self.is_on) is None:
^^^^^^^^^^
File "/config/custom_components/ids_hyyp/binary_sensor.py", line 108, in is_on
return bool(self.partition_data["zones"][self._zone_id]["triggered"])
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
KeyError: 'triggered'
2023-06-18 17:00:38.398 ERROR (MainThread) [homeassistant.components.binary_sensor] Error while setting up ids_hyyp platform for binary_sensor
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 328, in _async_setup_platform
await asyncio.gather(*pending)
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 471, in async_add_entities
await asyncio.gather(*tasks)
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 749, in _async_add_entity
await entity.add_to_platform_finish()
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 850, in add_to_platform_finish
self.async_write_ha_state()
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 590, in async_write_ha_state
self._async_write_ha_state()
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 654, in _async_write_ha_state
state = self._stringify_state(available)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 596, in _stringify_state
if (state := self.state) is None:
^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/components/binary_sensor/__init__.py", line 211, in state
if (is_on := self.is_on) is None:
^^^^^^^^^^
File "/config/custom_components/ids_hyyp/binary_sensor.py", line 108, in is_on
return bool(self.partition_data["zones"][self._zone_id]["triggered"])
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
KeyError: 'triggered'

**Comment by hawky358** [comment]
Thanks. 

> za.co.adt.securehome.android using adt ....

I believe this is the reason. The ADT is sort of an inherited legacy. I don't have an ADT system with which to test and get data. It may seems to be different and the API may have to be modified to work with the ADT version.

I'll see what I can do with this. I may need some other logs from the API to see the raw info, but I'll let you know.

**Comment by hawky358** [comment]
I've deployed a potential fix on the beta branch 1.3.2-b3.

- I've focused on the specific error that you've sent.  I'll take a look at the logs you've sent in more detail to see if I can spot something else.
- Please let me know what happens on your side.  If you get any more system level logs please send those to me, I've disabled the full logging on the integration for now.
- You will likely have a bunch of "Unavailable" sensors which you will have to manually delete. They will be called something like "[zone_name] Trigger".

**Comment by This-is-me-yes** [comment]
loaded and will let you know thx again

**Comment by hawky358** [comment]
Were you able to test the build?

**Comment by This-is-me-yes** [comment]
yes been working perfectly

**Comment by hawky358** [comment]
Bug addressed in 1.3.2
